### ISSUE TITLE
Consider lazy loading images

### ISSUE THREAD

**Issue by hangy** [openfoodfacts/openfoodfacts-server#1198]
Currently, all images are loaded directly. Especially for mobile users, lazy loading product images could improve perceived load time.

https://www.sitepoint.com/five-techniques-lazy-load-images-website-performance/

**Comment by stephanegigandet** [comment]
At least for me, it makes perceived load time worse: I see images appearing 2 seconds after the page loads, whereas they loaded instantly before, especially when they were in cache already. With lazyload, even if images are in cache, they don't start to appear before the onload, at 2.5 seconds in my browser.

I think we should use lazy load only for the big front + ingredients + nutrition full images that are in the modal dialog of the product page. Those are currently loaded even though most people won't see them.

e.g. on that one in the snippet below: https://static.openfoodfacts.org/images/products/376/012/296/1975/front_fr.11.full.jpg

```

<div class="row">
--
  | <div class="hide-for-large-up medium-12 columns"><div id="image_box_front" class="image_box" itemprop="image" itemscope itemtype="https://schema.org/ImageObject">
  | <a data-reveal-id="drop_front_fr" class="th">
  | <img id="og_image" class="hide-for-xlarge-up" src="https://static.openfoodfacts.org/images/products/376/012/296/1975/front_fr.11.200.jpg" srcset="https://static.openfoodfacts.org/images/products/376/012/296/1975/front_fr.11.400.jpg 2x" itemprop="image" width="149" height="200" alt="Palmiers allongés pur beurre aux gouttes de chocolat - Produit" itemprop="thumbnail" />
  | <img class="show-for-xlarge-up" src="https://static.openfoodfacts.org/images/products/376/012/296/1975/front_fr.11.400.jpg"  width="299" height="400" alt="Palmiers allongés pur beurre aux gouttes de chocolat - Produit" itemprop="thumbnail" />
  |  
  | </a>
  | <div id="drop_front_fr" class="reveal-modal" data-reveal aria-labelledby="modalTitle_front_fr" aria-hidden="true" role="dialog" about="https://static.openfoodfacts.org/images/products/376/012/296/1975/front_fr.11.full.jpg" >
  | <h2 id="modalTitle_front_fr"></h2>
  | <img src="https://static.openfoodfacts.org/images/products/376/012/296/1975/front_fr.11.full.jpg" alt="Palmiers allongés pur beurre aux gouttes de chocolat - Produit" itemprop="contentUrl" />
  | <a class="close-reveal-modal" aria-label="Close" href="#">&#215;</a>
  | <meta itemprop="representativeOfPage" content="false"/>
  | <meta itemprop="license" content="https://creativecommons.org/licenses/by-sa/3.0/"/>
  | <meta itemprop="caption" content="Palmiers allongés pur beurre aux gouttes de chocolat - Produit"/>
  | </div>

**Comment by stephanegigandet** [comment]
lazyload + the equalization of left column and main column makes part of the main column hidden. My guess is that the equalization happens before images are loaded, but when they are loaded, the size does not change...

An option would be to call \$(document).foundation('equalizer', 'reflow'); after an image is lazyloaded, but there's no callback.

As far as I recall, that was to make sure the left column's grey background extends all the way down.

**Comment by stephanegigandet** [comment]
The current javascript implementation of lazy loading has issues (I keep reloading pages in order to see the bottom of them - bug #1853) and the perceived speed is worse on desktop.

I would be in favour of removing the javascript lazyload. It seems that browsers will support it natively: https://addyosmani.com/blog/lazy-loading/

What do you think?

**Comment by hangy** [comment]
> The current javascript implementation of lazy loading has issues (I keep reloading pages in order to see the bottom of them - bug #1853) and the perceived speed is worse on desktop.

I don't experience that at all - load time is nearly instant (after a full refresh) for me on desktop.
![load](https://user-images.githubusercontent.com/87124/58817702-a10a4a80-862c-11e9-996e-09117cc00229.gif)
However, if it causes problems for some people, we should try doing something about it. :)

> I would be in favour of removing the javascript lazyload. It seems that browsers will support it natively: https://addyosmani.com/blog/lazy-loading/

Agreed. That's a good idea with the recent development. We should probably wait until the PR whatwg/html#3752 has been merged into some final spec.
